great_expectations/checkpoint/checkpoint.py

```diff
@@ -407,8 +408,41 @@ def _run_validation(
                 f"Exception occurred while running validation[{idx}] of Checkpoint '{self.name}': {e.message}."
             ) from e
 
+    @public_api
     def self_check(self, pretty_print=True) -> dict:
-        # Provide visibility into parameters that Checkpoint was instantiated with.
+        """
+        Method that is intended to provide visibility into parameters that Checkpoint was instantiated with.
+
+        If used as part of the test_yaml_config() workflow, the user will be able to know if the Checkpoint is
+        configured with all necessary parameters.
+
+        When run with self_check()::
```

Comment:
Is the double colon needed for a code block? For some reason the preview is not rendering the API docs to see how it renders.

Reply:
Checking now